plugins/ecobee/src/main.ts

```diff
@@ -0,0 +1,649 @@
+import axios, { AxiosRequestConfig } from 'axios'
+import sdk, { Device, DeviceInformation, ScryptedDeviceBase, OnOff, DeviceProvider, ScryptedDeviceType, ThermostatMode, Thermometer, HumiditySensor, TemperatureSetting, Settings, Setting, ScryptedInterface, Refresh, TemperatureUnit, HumidityCommand, HumidityMode, HumiditySetting } from '@scrypted/sdk';
+const { deviceManager, log } = sdk;
+
+const API_RETRY = 2;
+
+// Convert Fahrenheit to Celsius, round to 2 decimal places
+function convertFtoC(f: number) {
+  let c = (5/9) * (f - 32)
+  return Math.round(c*100)/100
+}
+
+// Convert Celsius to Fahrenheit, round to 1 decimal place
+function convertCtoF(c: number) {
+  let f = (c * 1.8) + 32
+  return Math.round(f*10)/10
+}
+
+function ecobeeToThermostatMode(mode: string) {
+  //  Values: auto, auxHeatOnly, cool, heat, off
+  switch(mode) {
+    case "cool":
+      return ThermostatMode.Cool;
+    case "heat":
+      return ThermostatMode.Heat;
+    case "auto":
+      return ThermostatMode.Auto;
+    case "off":
+      return ThermostatMode.Off;
+  }
+}
+
+function thermostatModeToEcobee(mode: ThermostatMode) {
+  //  Values: auto, auxHeatOnly, cool, heat, off
+  switch(mode) {
+    case ThermostatMode.Cool:
+      return "cool";
+    case ThermostatMode.Heat:
+      return "heat";
+    case ThermostatMode.Auto:
+      return "auto";
+    case ThermostatMode.Off:
+      return "off";
+  }
+}
+
+function humModeFromEcobee(mode: string): HumidityMode {
+  // Values: auto, manual, off
+  switch(mode) {
+    case 'auto':
+      return HumidityMode.Auto;
+    case "manual":
+      return HumidityMode.Humidify;
+  }
+
+  return HumidityMode.Off
+}
+
+class EcobeeThermostat extends ScryptedDeviceBase implements HumiditySensor, Thermometer, TemperatureSetting, Refresh, OnOff, HumiditySetting, Settings {
+  device: any;
+  revisionList: string[];
+  provider: EcobeeController;
+  on: boolean;
+
+  constructor(nativeId: string, provider: EcobeeController, info: DeviceInformation) {
+    super(nativeId);
+    this.provider = provider;
+    this.revisionList = null;
+    this.info = info;
+
+    this.temperatureUnit = TemperatureUnit.F
+    var modes: ThermostatMode[] = [ThermostatMode.Cool, ThermostatMode.Heat, ThermostatMode.Auto, ThermostatMode.Off];
```

Comment:
I generally `const` everything, and change it to a `let` as necessary. var is defecto deprecated due to weird block scope behavior.

Reply:
Good call. I think I actually copied that from what is likely some way old code in scrypted-nest 😀

I think const definitely makes sense though, so I'll swap that.